File: solo-leveling-ui-service/src/main/java/com/sleepkqq/sololeveling/ui/model/TgWebAppData.java
package com.sleepkqq.sololeveling.ui.model;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import java.time.LocalDateTime;

@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
public record TgWebAppData(
    LocalDateTime authDate,
    String chatType,
    String chatInstance,
    String hash,
    String signature,
    TgUserData user
) {

}

File: solo-leveling-ui-service/src/main/kotlin/com.sleepkqq.sololeveling/ui/model/TgWebAppData.kt
package com.sleepkqq.sololeveling.ui.model

import com.fasterxml.jackson.databind.PropertyNamingStrategies
import com.fasterxml.jackson.databind.annotation.JsonNaming
import java.time.LocalDateTime

@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy::class)
data class TgWebAppData(
	val authDate: LocalDateTime,
	val chatType: String?,
	val chatInstance: String?,
	val hash: String,
	val signature: String?,
	val user: TgUserData
)
